Makefile

```diff
@@ -5,90 +5,53 @@ PYTHONPATH := $(shell which python 2>/dev/null || which python3 2>/dev/null)
 
 SHELL=bash
 
-# Get absolute path of venv directory
-BASE_DIR := $(shell realpath .)
-VENV=$(BASE_DIR)/.venv
-ifeq ($(OS),Windows_NT)
-    VENV_BIN := $(VENV)/Scripts
-else
-    VENV_BIN := $(VENV)/bin
-endif
-
-# Define the path to main Python pyproject.toml
-PYPROJECT_PATH := python/quantum-pecos/pyproject.toml
-
 # Virtual environment
 # -------------------
 
 .PHONY: venv
-venv:  ## Build Python virtual environment
-	$(PYTHONPATH) -m venv $(VENV)
+venv:  ## Create virtual environment and install all dependencies
+	@echo "Installing uv..."
+	$(PYTHONPATH) -m pip install --upgrade uv
+	@echo "Creating venv and installing dependencies..."
+	uv sync
 
 # Requirements
 # ------------
-.PHONY: requirements
-requirements: metadeps installreqs  ## Install/refresh Python project requirements including those needed for development
 
 .PHONY: updatereqs
-updatereqs:  ## Auto update and generate requirements.txt
-	@echo "Upgrading pip..."
-	$(VENV_BIN)/python -m pip install --upgrade pip
-	$(VENV_BIN)/pip install -U pip-tools
-	-@rm python/quantum-pecos/requirements.txt
-	@echo "Using version: $(call get_version)"
-	@echo "Temporarily modifying pyproject.toml..."
-	@sed -i.bak '/pecos-rslib==/d' $(PYPROJECT_PATH)
-	$(VENV_BIN)/pip-compile --extra=tests --no-annotate --no-emit-index-url \
-		--output-file=python/quantum-pecos/requirements.txt \
-		--strip-extras $(PYPROJECT_PATH)
-	@echo "Restoring original pyproject.toml..."
-	@mv $(PYPROJECT_PATH).bak $(PYPROJECT_PATH)
-	@echo "Adding pecos-rslib back to requirements.txt..."
-	@echo "pecos-rslib==$(call get_version)" >> python/quantum-pecos/requirements.txt
-	@echo "numpy==1.26.4 ; python_version < \"3.13\"" >> python/quantum-pecos/requirements.txt
+updatereqs:  ## Generate/update lockfiles for both packages
+	@echo "Ensuring uv is installed..."
+	uv self update
+	@echo "Generating lock files..."
+	uv lock
 
 .PHONY: installreqs
-installreqs: upgrade-pip ## Install Python project requirements
-	@echo "Temporarily removing pecos-rslib from requirements..."
-	@grep -v "pecos-rslib" python/quantum-pecos/requirements.txt > temp_requirements.txt
-
-	@echo "Installing project requirements (excluding pecos-rslib)..."
-	$(VENV_BIN)/pip install -r temp_requirements.txt
-
-	@echo "Cleaning up temporary files..."
-	@rm temp_requirements.txt
-
-.PHONY: metadeps
-metadeps: upgrade-pip  ## Install extra dependencies used to develop/build this package
-	$(VENV_BIN)/pip install -U setuptools pip-tools pre-commit maturin
-
-# Function to extract version from pyproject.toml
-define get_version
-$(shell grep -m 1 'version\s*=' $(PYPROJECT_PATH) | sed 's/.*version\s*=\s*"\(.*\)".*/\1/')
-endef
+installreqs: ## Install Python project requirements to root .venv
+	@echo "Installing requirements..."
+	@unset CONDA_PREFIX && uv sync
```

Comment:
`@unset CONDA_PREFIX &&` shows up in a lot of places in this Makefile, why is it necessary? Even if it was set, isn't `uv` taking care of the correct paths?

Reply:
Probably. Will remove for now.